### ISSUE TITLE
Improve "expression name" documentation -- describe supported syntax

### ISSUE THREAD

**Issue by Eilon** [aspnet/Docs#2008]
_From @dougbu on August 14, 2014 14:50_

Find a page describing the Linq-to-objects expression name syntax that our `IHtmlHelper` implementation supports.  Link to that page from the appropriate parameter descriptions in our XML documentation.

See https://github.com/aspnet/Mvc/pull/961#discussion-diff-16214115 for background on this ask.

_Copied from original issue: aspnet/Mvc#1001_

**Comment by Eilon** [comment]
_From @dougbu on September 27, 2016 23:48_

This subject will be more accessible (i.e. easier for customers to find) in documentation and not doc comments. Suggest moving this issue to the [aspnet/docs](https://github.com/aspnet/docs) repo and marking that new issue as blocked on aspnet/Docs#131. aspnet/Docs#131 is about filling in https://docs.asp.net/en/latest/mvc/views/html-helpers.html but unfortunately it's on the backlog. A need to document supported expression name formats could also be part of aspnet/Docs#131 rather than a separate issue.

(Hmm, one shortcut to addressing aspnet/Docs#131 would be to rewrite MVC 4.x or 5.x documentation in latest markdown.)

Right now, the http://docs.asp.net site and [aspnet/docs](https://github.com/aspnet/docs) repo do not contain much information about `string`-based HTML helpers. The [Working with Forms](https://docs.asp.net/en/latest/mvc/views/working-with-forms.html) page does include an [Expression names](https://docs.asp.net/en/latest/mvc/views/working-with-forms.html#expression-names) sub-section but that comes at expression names from the lambda expression -> generated HTML `name` attribute perspective. Good news is that angle is almost completely covered. All that's missing is that `asp-for="CollectionProperty[23].Member"` and `asp-for="CollectionProperty[twentyThree].Member"` generate the same name when there's a `twentyThree` variable with the value `23`.

**Comment by Rick-Anderson** [comment]
> one shortcut to addressing aspnet/Docs#131 would be to rewrite MVC 4.x or 5.x documentation in latest markdown.)

@dougbu MVC 4.x and 5.x are retired and not being maintained.

@danroth27  might have an idea.

**Comment by Rick-Anderson** [comment]
>The [Working with Forms](https://docs.asp.net/en/latest/mvc/views/working-with-forms.html) page does include an [Expression names](https://docs.asp.net/en/latest/mvc/views/working-with-forms.html#expression-names) sub-section but that comes at expression names from the lambda expression -> generated HTML `name` attribute perspective. Good news is that angle is almost completely covered. All that's missing is that `asp-for="CollectionProperty[23].Member"` and `asp-for="CollectionProperty[twentyThree].Member"` generate the same name when there's a `twentyThree` variable with the value `23`.

@ardalis  can you do this?
